x-pack/plugins/session_view/public/components/tty_player/hooks.test.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { renderHook, act } from '@testing-library/react-hooks';
+import { sessionViewIOEventsMock } from '../../../common/mocks/responses/session_view_io_events.mock';
+import { useIOLines, useXtermPlayer, XtermPlayerDeps } from './hooks';
+import { ProcessEventsPage } from '../../../common/types/process_tree';
+
+// https://stackoverflow.com/questions/39830580/jest-test-fails-typeerror-window-matchmedia-is-not-a-function
+// xtermjs is using window.matchMedia, which isn't mocked in jest by default.
+Object.defineProperty(window, 'matchMedia', {
+  writable: true,
+  value: jest.fn().mockImplementation((query) => ({
+    matches: false,
+    media: query,
+    onchange: null,
+    addListener: jest.fn(), // Deprecated
+    removeListener: jest.fn(), // Deprecated
+    addEventListener: jest.fn(),
+    removeEventListener: jest.fn(),
+    dispatchEvent: jest.fn(),
+  })),
+});
```

Comment:
To avoid side-effects on other tests, it's better to wrap this on a `beforeAll`. (probably there are no other tests using `matchMedia, but who knows...)

```js
beforeAll(() => {
// ...
});
```

Reply:
good call.

Reply:
So, window.matchMedia is undefined in the jest dom, so it shouldn't effect any other tests. I could put it in the beforeAll, and then in afterAll, just set window.matchMedia back to 'undefined'..